### PR TITLE
test: 80% coverage of unixfs/io [WIP]

### PR DESCRIPTION
**Pull request by Kubuxu**
License: MIT
Signed-off-by: Jakub Sztandera <kubuxu@protonmail.ch>

### CODE REVIEW

File: unixfs/io/dagserv_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/ipfs/go-ipfs/importer/chunk"
+	mdag "github.com/ipfs/go-ipfs/merkledag"
+
+	ds "gx/ipfs/QmTxLSvdhwg68WJimdS6icLPhZi28aTp6b7uihC2Yb47Xk/go-datastore"
+	"gx/ipfs/QmTxLSvdhwg68WJimdS6icLPhZi28aTp6b7uihC2Yb47Xk/go-datastore/sync"
+	u "gx/ipfs/QmZNVWh8LLjAavuQ2JXuFmuYH3C11xo988vSgp7UQrTRj1/go-ipfs-util"
+	context "gx/ipfs/QmZy2y8t9zQH2a1b8q2ZSLKp17ATuJoCNxxyMFG5qFExpt/go-net/context"
+)
+
+func getMockDagServ(t testing.TB) mdag.DAGService {
```
Comment (whyrusleeping) [inline unixfs/io/dagserv_test.go]: i would just use `merkledag/test.Mock()` for this

**Comment by whyrusleeping** [comment]
maybe some of this testing utility code can be combined into a package and reused across different test suites

File: unixfs/io/dagserv_test.go (hunk 1 above)
Comment (Kubuxu) [inline unixfs/io/dagserv_test.go]: Thanks.

**Comment by Kubuxu** [comment]
Yeah, I will do that. I just wanted to get first test case up.

**Comment by whyrusleeping** [comment]
@Kubuxu have an example of a cyclic import issue?

**Comment by Kubuxu** [comment]
If we wanted to extract it outside of the unixfs package, otherwise it will be good.

**Comment by whyrusleeping** [comment]
@Kubuxu is this ready for review? title still has WIP

**Comment by Kubuxu** [comment]
It is still WIP, it is at about 70% right now.

**Comment by Kubuxu** [comment]
```
=== RUN   TestBasicRead
--- PASS: TestBasicRead (0.00s)
=== RUN   TestSeekAndRead
--- PASS: TestSeekAndRead (0.00s)
=== RUN   TestRelativeSeek
--- PASS: TestRelativeSeek (0.01s)
=== RUN   TestTypeFailures
--- PASS: TestTypeFailures (0.00s)
=== RUN   TestBadPBData
--- PASS: TestBadPBData (0.00s)
=== RUN   TestMetadataNode
--- PASS: TestMetadataNode (0.00s)
=== RUN   TestWriteTo
--- PASS: TestWriteTo (0.00s)
=== RUN   TestReaderSzie
--- PASS: TestReaderSzie (0.00s)
=== RUN   TestEmptyNode
--- PASS: TestEmptyNode (0.00s)
=== RUN   TestDirBuilder
--- PASS: TestDirBuilder (0.00s)
PASS
coverage: 81.7% of statements
ok  	github.com/ipfs/go-ipfs/unixfs/io	1.026s
```

**Comment by whyrusleeping** [comment]
@Kubuxu gotta rebase this one

**Comment by Kubuxu** [comment]
Rebased

**Comment by whyrusleeping** [comment]
LGTM, thanks!

**Pull request merged**